src/planner/vectorfield/VectorFieldUtil.cpp

```diff
@@ -0,0 +1,155 @@
+    problem.addConstantConstraint(1, 0, nextPosition - currentPosition);
+    problem.addConstantConstraint(1, 1, nextVelocity);
+    const auto solution = problem.fit();
+    const auto coefficients = solution.getCoefficients().front();
+
+    _stateSpace->expMap(currentPosition, currState);
+    _outputTrajectory->addSegment(coefficients, segmentDuration, currState);
+  }
+  return _outputTrajectory;
+}
+
+//==============================================================================
+
+DesiredTwistFunction::DesiredTwistFunction(
+    const Twist& _twist, const Jacobian& _jacobian)
+  : dart::optimizer::Function("DesiredTwistFunction")
+  , mTwist(_twist)
+  , mJacobian(_jacobian)
+{
+}
+
+//==============================================================================
+
+double DesiredTwistFunction::eval(const Eigen::VectorXd& _qd)
+{
+  return 0.5 * (mJacobian * _qd - mTwist).squaredNorm();
+}
+
+//==============================================================================
+
+void DesiredTwistFunction::evalGradient(
+    const Eigen::VectorXd& _qd, Eigen::Map<Eigen::VectorXd> _grad)
+{
+  _grad = mJacobian.transpose() * (mJacobian * _qd - mTwist);
+}
+
+//==============================================================================
+
+bool computeJointVelocityFromTwist(
+    const Eigen::Vector6d& _desiredTwist,
+    const aikido::statespace::dart::MetaSkeletonStateSpacePtr _stateSpace,
+    const dart::dynamics::BodyNodePtr _bodyNode,
+    double _optimizationTolerance,
+    double _timestep,
+    double _padding,
+    Eigen::VectorXd* _jointVelocity)
+{
+  using dart::math::Jacobian;
+  using dart::optimizer::Problem;
+  using dart::optimizer::Solver;
+  using Eigen::VectorXd;
+  // Use LBFGS to find joint angles that won't violate the joint limits.
+  const Jacobian jacobian
+      = _stateSpace->getMetaSkeleton()->getWorldJacobian(_bodyNode);
+
+  const std::size_t numDofs = _stateSpace->getMetaSkeleton()->getNumDofs();
+  VectorXd lowerLimits(numDofs);
+  VectorXd upperLimits(numDofs);
+
+  VectorXd positions = _stateSpace->getMetaSkeleton()->getPositions();
+  VectorXd positionLowerLimits
+      = _stateSpace->getMetaSkeleton()->getPositionLowerLimits();
+  VectorXd positionUpperLimits
+      = _stateSpace->getMetaSkeleton()->getPositionUpperLimits();
+  VectorXd velocityLowerLimits
+      = _stateSpace->getMetaSkeleton()->getVelocityLowerLimits();
+  VectorXd velocityUpperLimits
+      = _stateSpace->getMetaSkeleton()->getVelocityUpperLimits();
+
+  for (std::size_t i = 0; i < numDofs; ++i)
+  {
+    const double position = positions[i];
+    const double positionLowerLimit = positionLowerLimits[i];
+    const double positionUpperLimit = positionUpperLimits[i];
+    const double velocityLowerLimit = velocityLowerLimits[i];
+    const double velocityUpperLimit = velocityUpperLimits[i];
+
+    if (position
+        < positionLowerLimit - _timestep * velocityLowerLimit + _padding)
+      lowerLimits[i] = 0;
```

Comment:
I think this restriction can be a little bit relaxed. Current logic just sets the velocity to zero when the integrated position with the velocity exceeds the padded position limit. However, if the current position doesn't yet exceed the padded position limit, then it would be fine to integrate the position towards the padded position limit with the proper velocity, which might be less than the velocity limit.

I propose to change the logic as
```cpp
lowerLimits[i] = std::max(velocityLowerLimits[i], ((velocityLowerLimits[i] + _padding) - position) / _timestep);
```
Similarly, below can be changed as
```cpp
upperLimits[i] = std::min(velocityUpperLimits[i], ((velocityUpperLimits[i] - _padding) - position) / _timestep);
```

One (positive I think) side effect is that the new logic will compute `lower/upperLimits` that will fix the current position limit violations if there are.